### PR TITLE
Support Mellanox-SN4600C-C64 as T1 switch in dual-ToR scenario

### PR DESCRIPTION
**Pull request by stephenxs**
#### Why I did it

Support Mellanox-SN4600C-C64 as T1 switch in dual-ToR scenario
This is to port #11032 and #11299 from 202012 to master.
1. Support additional queue and PG in buffer templates, including both traditional and dynamic model
2. Support mapping DSCP 2/6 to lossless traffic in the QoS template.
3. Add macros to generate additional lossless PG in the dynamic model
4. Adjust the order in which the generic/dedicated (with additional lossless queues) macros are checked and called to generate buffer tables in common template `buffers_config.j2`
   - Buffer tables are rendered via using macros.
   - Both generic and dedicated macros are defined on our platform. Currently, the generic one is called as long as it is defined, which causes the generic one always being called on our platform. To avoid it, the dedicated macrio is checked and called first and then the generic ones.
5. Support MAP_PFC_PRIORITY_TO_PRIORITY_GROUP on ports with additional lossless queues.

On Mellanox-SN4600C-C64, buffer configuration for t1 is calculated as:
- 40 * 100G downlink ports with 4 lossless PGs/queues, 1 lossy PG, and 3 lossy queues
- 16 * 100G uplink ports with 2 lossless PGs/queues, 1 lossy PG, and 5 lossy queues

Signed-off-by: Stephen Sun <stephens@nvidia.com>

#### How to verify it

Run regression test.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205

### CODE REVIEW

**Review by baiwei0427 — APPROVED** [review]

**Comment by stephenxs** [comment]
To cleanly cherry-pick to 202205 it depends on #11448

File: files/build_templates/buffers_config.j2 [hunk 1]
```diff
@@ -168,8 +168,8 @@ def
 {{ defs.generate_profile_lists_with_inactive_ports(port_names_active, port_names_inactive) }},
 {% endif %}
 
-{%- if defs.generate_pg_profils is defined %}
-{{ defs.generate_pg_profils(port_names_active) }}
```
[the remark this answers is not in the document]
Comment (stephenxs) [reply files/build_templates/buffers_config.j2:172]: Fixed

**Comment by stephenxs** [comment]
Previously it failed because t1-lag timeout is reached. #11478 to fix it.

**Review by neethajohn — APPROVED** [review]

**Pull request merged**